sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -449,5 +481,49 @@ internal static bool GlobalDisableAutoRecording
                 return disableAutoRecording || GlobalIsRunningInCI;
             }
         }
+
+        private void BootStrapTestResources()
+        {
+            try
+            {
+                _bootstrappingSemaphore.Wait();
```

Comment:
Lock?

Reply:
Should we test `_bootstrappingAttempted` before grabbing the semaphore to avoid paying the cost if we know it was already bootstrapped?